### DISCUSSION TITLE
Support `generateStaticParams`

### DISCUSSION THREAD

**Discussion by minyan-gu** [vinissimus/next-translate#1098] [General]
Am I right this package does not support `generateStaticParams` amt?

When I run `next build` on latest Next.js version `13.4.7`, it shows me build errors:
```
Error occurred prerendering page "/en/blog". Read more: https://nextjs.org/docs/messages/prerender-error
Error: The provided export path '/blog' doesn't match the '/[lang]/blog/page' page.
...
Export encountered errors on following paths:
        /[lang]/blog/page: /en/blog
        /[lang]/blog/page: /es/blog
```

Can you please confirm this is the current expected behaviour? If so, is there any plan to support `generateStaticParams` for static generation at build time?

Thanks.

**Comment by aralroca** [comment]
It should work. Please write more info about how is your `generateStaticParams`, otherwise we don't have context to help you.

**Reply by minyan-gu** [reply]
Sure, here's my current `app` structure:
```bash
app
├── [lang]
│   ├── blog
│   │   └── page.tsx
│   └── layout.tsx
```

1. `/[lang]/layout.tsx` as the root layout:
```tsx
export default function RootLayout({
	children,
	params,
}: {
	children: React.ReactNode;
	params: { lang: string };
}) {
	const { lang } = params;

	return (
		<html lang={lang} className={`${poppins.variable} ${noto_sans_mono.variable}`}>
			<body>
				<Providers>{children}</Providers>
			</body>
		</html>
	);
}
```

`Providers` here is a React context, which uses `use client` at the top, so the Context is a client component.

2. The blog index page `/[lang]/blog/page.tsx`
```tsx
const languages = ['en', 'es'];

export async function generateStaticParams() {
	return languages.map(lang => ({ lang }));
}

export default function Page({
	params,
}: {
	params: {
		lang: string;
	};
}) {
	const { t } = useTranslation('common');

	return (
		<div className="flex">
			<h1>{t`title`}</h1>
			<h2>{t('variable-example', { count: 11 })}</h2>
		</div>
	);
}
```

3. `next.config.js`
```js
/** @type {import('next').NextConfig} */
// eslint-disable-next-line @typescript-eslint/no-var-requires
const nextTranslate = require('next-translate-plugin');

module.exports = nextTranslate({
	basePath: '/alpha',
	compress: false,
});
```

4. `i18n.js`
```js
module.exports = {
	locales: ['en', 'es'],
	defaultLocale: 'en',
	pages: {
		'*': ['common'],
		// '/[lang]/blog': ['blog'],
	},
};
```

I think that's all. if you need anything else, please let me know. Thanks.

**Comment by minyan-gu** [comment]
Feels like sth similar to this? `https://github.com/vercel/next.js/issues/46622`

**Comment by minyan-gu** [comment]
Would be helpful if you add one working example  using `generateStaticParams`, within your `with-app-dir` app:

https://github.com/aralroca/next-translate/blob/de5ad45d005488b36f21fed291e18437af468972/examples/with-app-directory/src/app/%5Blang%5D/second-page/page.tsx

**Reply by aralroca** [reply]
It's nothing we can do, is out of the scope of our library, and probably should be fixed after Next.js fix this issue https://github.com/vercel/next.js/issues/46622
